kernel/src/hil/i2c.rs

```diff
@@ -113,6 +181,69 @@ pub trait I2CDevice {
     fn read(&self, buffer: &'static mut [u8], len: u8);
 }
 
+pub trait SMBusDevice: I2CDevice {
+    /// Write data then read data to a slave device in an SMBus
+    /// compatible way.
+    ///
+    /// This function will use the I2C master to write data to a device and
+    /// then read data from the device in a SMBus compatible way. This will be
+    /// a best effort attempt to match the SMBus specification based on what
+    /// the hardware can support.
+    /// This function is expected to make any hardware changes required to
+    /// support SMBus and then revert those changes to support future I2C.
+    ///
+    /// data: The buffer to write the data from and read back to
+    /// write_len: The length of the write operation
+    /// read_len: The length of the read operation
+    #[allow(unused_variables)]
+    fn smbus_write_read(
+        &self,
+        data: &'static mut [u8],
+        write_len: u8,
+        read_len: u8,
+    ) -> Result<(), (Error, &'static mut [u8])> {
+        Err((Error::NotSupported, data))
+    }
```

Comment:
why provide a default implementation for these? There are already implementations on the SMBusDevice struct in virtual_i2c.rs, and normal I2CDevices don't have to implement this trait anyway.

Reply:
Removed